src/utils/delegation.ts

```diff
@@ -0,0 +1,88 @@
+import { subgraphRequest } from '../utils';
+import delegationSubgraphs from '../delegationSubgraphs.json';
+
+export const SNAPSHOT_SUBGRAPH_URL = delegationSubgraphs;
+
+type Delegation = {
+  delegator: string;
+  delegate: string;
+  space: string;
+  timestamp: number;
+};
+
+export default async function getDelegatesBySpace(
+  network: string,
+  space: string,
+  snapshot: string | number = 'latest',
+  options: any = {}
+) {
+  const subgraphUrl = options.subgraphUrl || SNAPSHOT_SUBGRAPH_URL[network];
+  if (!subgraphUrl) {
+    return Promise.reject(
+      `Delegation subgraph not available for network ${network}`
+    );
+  }
+  const spaceIn = ['', space];
+  if (space.includes('.eth')) spaceIn.push(space.replace('.eth', ''));
+
+  const PAGE_SIZE = 1000;
+  let result: Delegation[] = [];
+  const params: any = {
+    delegations: {
+      __args: {
+        where: {
+          space_in: spaceIn
+        },
+        first: PAGE_SIZE,
+        skip: 0,
+        orderBy: 'timestamp',
+        orderDirection: 'asc'
+      },
+      delegator: true,
+      space: true,
+      delegate: true,
+      timestamp: true
+    }
+  };
+  if (snapshot !== 'latest') {
+    params.delegations.__args.block = { number: snapshot };
+  }
+
+  while (true) {
+    params.delegations.__args.where.timestamp_gte =
+      result[result.length - 1]?.timestamp || 0;
+
+    const newResults: Delegation[] =
+      (await subgraphRequest(subgraphUrl, params)).delegations || [];
```

Comment:
Very highly unlikely but, could we return an error if all delegations of a page have same timestamp 😄

Reply:
Implemented, now handling case when there are more than 1000 duplicates, it should be able to handle up to 5000 delegations with same timestamp